modules/viewabilityScoreGeneration.js

```diff
@@ -15,10 +16,64 @@ const GPT_SLOT_RENDER_ENDED_EVENT = 'slotRenderEnded';
 const GPT_IMPRESSION_VIEWABLE_EVENT = 'impressionViewable';
 const GPT_SLOT_VISIBILITY_CHANGED_EVENT = 'slotVisibilityChanged';
 const TOTAL_VIEW_TIME_LIMIT = 1000000000;
-const domain = window.location.hostname;
 const NATIVE_DEFAULT_SIZE = '0x0';
+const DEFAULT_SERVER_CALL_FREQUENCY = { metric: 'hours', duration: 6 };
 const ADSLOTSIZE_INDEX = 2;
 const ADUNIT_INDEX = 1;
+const ENDPOINT = 'http://localhost:3000/fake-endpoint/inventory-packaging' // TODO: update this to a real endpoint when ready
+const domain = window.location.hostname;
+let vsgObj;
+
+storage.getDataFromLocalStorage('viewability-data', val => {
+  vsgObj = JSON.parse(val);
+});
+
+export const fireToServer = auctionData => {
+  const payload = generatePayload(auctionData, vsgObj);
+
+  ajax(ENDPOINT, (res) => {
+    // do nothing for now
+  }, JSON.stringify(payload));
+
+  vsgObj = null;
+  removeFromLocalStorage('viewability-data');
+};
+
+export const generatePayload = (auctionData, lsObj) => {
+  const adSizeKeys = Object.keys(lsObj).filter(key => /([0-9]+x[0-9]+)/.test(key));
+  const adUnitKeys = Object.keys(lsObj).filter(key => !/([0-9]+x[0-9]+)/.test(key) && key !== window.location.hostname);
+
+  return {
+    iid: auctionData.auctionId,
+    recordTs: Date.now(),
+    pubid: getPubId(auctionData),
+    adDomain: window.location.hostname,
+    adUnits: adUnitKeys.map(adUnitKey => {
+      lsObj[adUnitKey].adUnit = adUnitKey;
+      return lsObj[adUnitKey];
+    }),
+    adSizes: adSizeKeys.map(adSizeKey => {
+      lsObj[adSizeKey].adSize = adSizeKey;
+      return lsObj[adSizeKey];
+    })
+  };
+};
+
+const getPubId = auctionData => {
```

Comment:
Do we need this function? Or can we read this from PWT ?

Reply:
Ah good question, I doubled checked if we could take this from PWT, but looks like I wasn't able to find the publisher id on there.  So I guess this function would be ok to keep.